lib/src/sql/ml_model.rs

```diff
@@ -0,0 +1,133 @@
+use async_recursion::async_recursion;
+use derive::Store;
+use nom::{
+	bytes::complete::{tag, take_while1},
+	character::complete::i64,
+	combinator::{cut, recognize},
+	multi::separated_list1,
+};
+use revision::revisioned;
+use serde::{Deserialize, Serialize};
+use std::fmt;
+
+use crate::{
+	ctx::Context,
+	dbs::{Options, Transaction},
+	doc::CursorDoc,
+	err::Error,
+	sql::{error::IResult, value::Value},
+};
+
+use super::{
+	common::{closechevron, closeparentheses, openchevron, openparentheses, val_char},
+	error::{expect_tag_no_case, expected},
+	util::expect_delimited,
+	value::value,
+};
+
+#[derive(Clone, Debug, Default, PartialEq, PartialOrd, Serialize, Deserialize, Store, Hash)]
+#[revisioned(revision = 1)]
+pub struct MlModel {
+	pub name: String,
+	pub version: String,
+	pub parameters: Value,
+}
+
+impl fmt::Display for MlModel {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		write!(f, "ml::{}<{}>({})", self.name, self.version, self.parameters)
+	}
+}
+
+impl MlModel {
+	#[cfg_attr(not(target_arch = "wasm32"), async_recursion)]
+	#[cfg_attr(target_arch = "wasm32", async_recursion(?Send))]
+	pub(crate) async fn compute(
+		&self,
+		_ctx: &Context<'_>,
+		_opt: &Options,
+		_txn: &Transaction,
+		_doc: Option<&'async_recursion CursorDoc<'_>>,
+	) -> Result<Value, Error> {
+		Err(Error::Unimplemented("Ml model evaluation not yet implemented".to_string()))
```

Comment:
TODO: We should add a serialiser for this too at some point before or when we implement it.